examples/water/water_shader.frag

```diff
@@ -0,0 +1,41 @@
+#version 450
+
+const vec3 water_colour = vec3(0.0, 117.0 / 255.0, 242.0 / 255.0);
+const float zNear = 10.0;
+const float zFar = 400.0;
+
+layout(set = 0, binding = 1) uniform texture2D reflection;
+layout(set = 0, binding = 2) uniform texture2D terrain_depth_tex;
+layout(set = 0, binding = 3) uniform sampler colour_sampler;
+
+layout(location = 0) in PerVertex {
+    vec2 f_WaterScreenPos;
+    float f_Fresnel;
+    vec3 f_Light;
+} f_In;
+
+layout(location = 0) out vec4 outColor;
+
+float to_linear_depth(float depth) {
+    float z_n = 2.0 * depth - 1.0;
+    float z_e = 2.0 * zNear * zFar / (zFar + zNear - z_n * (zFar - zNear));
+    return z_e;
+}
+
+void main() {
+    vec3 reflection_colour = texture(sampler2D(reflection, colour_sampler), f_In.f_WaterScreenPos.xy).xyz;
+
+    float my_depth = to_linear_depth(gl_FragCoord.z);
+    float terrain_depth = to_linear_depth(texture(sampler2D(terrain_depth_tex, colour_sampler), gl_FragCoord.xy / vec2(1024.0, 768.0)).r);
```

Comment:
so this 1024x768 will become wrong as soon as we resize
can we pass it as a uniform constant?